FormalConjectures/ForMathlib/Combinatorics/Basic.lean

```diff
@@ -20,19 +20,13 @@ import Mathlib.Order.Defs.PartialOrder
 
 open Function Set
 
-/-- A Sidon sequence (also called a Sidon set) is a sequence of natural numbers $a_0, a_1, ...$
-such that all pairwise sums $a_i + a_j$ are distinct for $i ≤ j$.
+variable {α γ: Type*} [AddCommMonoid α] [Membership α γ]
 
-In other words, all pairwise sums of elements are distinct apart from coincidences forced by the
-commutativity of addition. -/
-def IsSidon {ι α : Type*} [Preorder ι] [Preorder α] [Add α] (a : ι →o α) : Prop :=
-  ∀ i₁ j₁ i₂ j₂, i₁ ≤ j₁ → i₂ ≤ j₂ → a i₁ + a j₁ = a i₂ + a j₂ → i₁ = i₂ ∧ j₁ = j₂
-
-lemma IsSidon.injective {ι α : Type*} [Preorder ι] [Preorder α] [Add α] (a : ι →o α) (ha : IsSidon a) :
-    Injective a :=
-  fun i₁ i₂ hi ↦ (ha i₁ i₁ i₂ i₂ (le_refl _) (le_refl _) (by rw [hi])).1
-
-variable {α : Type*} [AddCommMonoid α]
+/-- A Sidon set is a set, such that such that all pairwise sums of elements are distinct apart from
+coincidences forced by the commutativity of addition. -/
+def IsSidon (A : γ) : Prop :=
+  Pairwise fun i₁ j₁ =>
+  i₁ ∈ A → j₁ ∈ A → ∀ i₂ ∈ A, ∀ j₂ ∈ A, i₁ + i₂ = j₁ + j₂ → (i₁ = j₂ ∧ i₂ = j₁)
```

Comment:
This formulation is a bit surprising to me, I expected the more symmetric version (for all i1 j1 i2 j2 in A, i1 + i2 = j1 + j2 implies (i1=j1 and i2=j2) or (i1=j2 and i2=j1). I don't mind this one, but we should then have the lemma saying they're equivalent. Note also you could use Set.Pairwise with A.

Reply:
done, changed it to the symmetric version....